### PR TITLE
Documentation and Package Restructure

### PR DESCRIPTION
**Pull request by adamgayoso**
- Adds documentation for DoubletDetection via Readthedocs

Restructures package in a more sensible way
- Moves requirements into `setup.py` so they get installed automatically
- Unfortunately now `phenograph` has to be manually installed (it basically was before wrapped in requirements.txt)
- If `phenograph` is not installed, an `ImportError` is raised with install instructions.

Note that `phenograph` will become optional in v3.0 so this is a fine temporary solution.

### CODE REVIEW

**Comment by adamgayoso** [comment]
The new package structure was tested. Everything imports fine.

**Comment by JonathanShor** [comment]
Why should this not go into 3.0?

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,127 @@
-*.pyc
```
Comment (JonathanShor) [inline .gitignore]: I appreciate your comment that this kitchen-sink style gitignore is used by other packages, but IMO it shows a lack of control or understanding of your own package. Having git ignore miscellaneous filetypes just serves to hide issues within your own local repo. How does this help?

If there are some unexpected Django (?!) files lurking in the repo, I **want** to see them.

Additionally, the version-control ignore is the wrong place for local setup ignores, i.e. .vscode. Stuff like that belongs in either your system-wide ignore (`~/.config/git/ignore`), or in the repo-wide ignore (`.git/info/exclude`), or in a .gitignore inside the offending folder (`.vscode/.gitignore`).
Imposing your local setup concerns on the public repo seems unnecessary.

File: setup.py [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
 import sys
 
 CLASSIFIERS = [
     "Development Status :: 4 - Beta",
     "Natural Language :: English",
     "License :: OSI Approved :: MIT",
     "Operating System :: OS Independent",
-    "Programming Language :: Python :: 3.5",
-    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
```
Comment (JonathanShor) [inline setup.py]: What's driving this change, specifically in this PR?

We likely need it regardless for 3.0, especially if we will completely type-hint, but let's document the reason here.

File: setup.py [hunk 3]
```diff
@@ -1,18 +1,17 @@
-from setuptools import setup
+from setuptools import setup, find_packages
```
Comment (JonathanShor) [inline setup.py]: Reason for this? It's not like we have multiple packages.

File: setup.py [hunk 4]
```diff
@@ -23,7 +22,17 @@
     url='https://github.com/JonathanShor/DoubletDetection',
     author='Adam Gayoso, Jonathan Shor, Ambrose J. Carr',
     author_email='ajg2188@columbia.edu',
-    package_dir={'': 'src'},
```
Comment (JonathanShor) [inline setup.py]: Rationale for the move from `src`?

File: README.md [hunk 5]
```diff
@@ -1,16 +1,18 @@
...
+
 
 DoubletDetection is a Python3 package to detect doublets (technical errors) in single-cell RNA-seq count matrices.
 
 
 To install DoubletDetection:
 
 ```
+pip3 install git+https://github.com/JonathanShor/PhenoGraph.git
 git clone https://github.com/JonathanShor/DoubletDetection.git
```
[the remark this answers is not in the document]
Comment (adamgayoso) [reply README.md]: Not sure I understand this question. It's hard to put a github repository in the setup.py, so this is a temporary solution until 3.0 when we remove PhenoGraph dependency.

File: setup.py (hunk 4 above)
Comment (adamgayoso) [reply setup.py]: It seems pretty standard to not have an empty src folder just hanging out. Most projects use this arrangement.

For example - https://github.com/numpy/numpy, or pytorch, or Scanpy, etc.

File: setup.py (hunk 2 above)
Comment (adamgayoso) [reply setup.py]: I'll add back python 3.6 as I'm not sure we need to delete this for this PR

File: .gitignore (hunk 1 above)
Comment (adamgayoso) [reply .gitignore]: I resolved this by adding .vscode to .git/info/exclude and removing excess in `.gitignore`.

File: setup.py (hunk 3 above)
Comment (adamgayoso) [reply setup.py]: Not a good reason as we only have one package, but it's general enough and doing no harm and it works :)

**Comment by adamgayoso** [comment]
> Why should this not go into 3.0?

This doesn't change the API, but really just adds documentation. The documentation is relevant now, as opposed to when 3.0 comes out.

**Comment by adamgayoso** [comment]
By the way, we can squash this into master.

**Comment by adamgayoso** [comment]
> Also, I'm not familiar with sphinx, so I'm assuming your design is reasonable re: Makefile etc. Being self contained within the docs folder seems good.

Sphinx pretty much automates this.

File: setup.py (hunk 3 above)
Comment (JonathanShor) [reply setup.py]: The exact same rationale explains why NOT to do this.... but ¯\_(ツ)_/¯

**Pull request merged**